### PR TITLE
Update subscribed communities in sidebar when user subscribes

### PR DESCRIPTION
**Pull request by twizmwazin**

### CODE REVIEW

File: app/src/main/java/com/jerboa/ui/components/community/CommunityActivity.kt [hunk 1]
```diff
@@ -235,6 +236,11 @@ fun CommunityActivity(
                                                             auth = acct.jwt,
                                                         ),
                                                     )
+                                                    siteViewModel.getSite(
```
Comment (MV-GH) [inline app/src/main/java/com/jerboa/ui/components/community/CommunityActivity.kt:239]: Does this guarantee that getSite only is executed once followCommunity is done?

Comment (twizmwazin) [reply app/src/main/java/com/jerboa/ui/components/community/CommunityActivity.kt]: Fixed!

File: app/src/main/java/com/jerboa/ui/components/community/CommunityViewModel.kt [hunk 2]
```diff
@@ -144,6 +145,8 @@ class CommunityViewModel : ViewModel(), Initializable {
 
                 else -> {}
             }
+
+            onComplete()
```
Comment (MV-GH) [inline app/src/main/java/com/jerboa/ui/components/community/CommunityViewModel.kt:149]: I would only do it on Success, there is no point on updating if the following has failed

File: app/src/main/java/com/jerboa/ui/components/community/CommunityActivity.kt [hunk 3]
```diff
@@ -234,6 +235,13 @@ fun CommunityActivity(
                                                             follow = cfv.subscribed == SubscribedType.NotSubscribed,
                                                             auth = acct.jwt,
                                                         ),
+                                                        onComplete = {
```
Comment (dessalines) [inline app/src/main/java/com/jerboa/ui/components/community/CommunityActivity.kt:238]: Nice, this is simple enough and works.

**Review by MV-GH — APPROVED** [review]

**Pull request merged**